tests/std/tests/P0896R4_common_iterator/test.cpp

```diff
@@ -311,6 +311,31 @@ constexpr bool test_lwg_3574() {
     return true;
 }
 
+// Validate that _Variant works when fed with a non-trivially-destructible type
+struct non_trivially_destructible_input_iterator {
+    constexpr ~non_trivially_destructible_input_iterator() {}

```

Comment:
(Pure style, ignore if you like) Since this class is used only in the `COMPILE-ONLY` test case below, you could remove the `constexpr`s and the function bodies - it's all extraneous. You could also define this class locally in `test_non_trivially_destructible_type` since it isn't used outside.

Reply:
It seems that the function bodies are needed for the programs to link. Other suggested changes are applied.

Reply:
> It seems that the function bodies are needed for the programs to link. Other suggested changes are applied.

Yes, sorry - this function is actually link-only. 3AM Casey is easily confused.